mono/sgen/sgen-gc.h

```diff
@@ -827,7 +827,10 @@ void sgen_null_link_in_range (int generation, ScanCopyContext ctx, gboolean trac
 void sgen_process_fin_stage_entries (void)
 	MONO_PERMIT (need (sgen_gc_locked));
 gboolean sgen_have_pending_finalizers (void);
-void sgen_object_register_for_finalization (GCObject *obj, void *user_data)
+
+typedef void (*SGenFinalizationProc)(void*, void*); // same as MonoFinalizationProc, GC_finalization_proc
```

Comment:
Use `gpointer`.

Reply:
The point is to have correct types where it is easy/possible and avoid casts. In this case it is easy. The pointers are always of a precise type and the casts can be removed.

Reply:
@luhenry you mean `typedef void (*SGenFinalizationProc)(gpointer, gpointer);` right?

Reply:
Oh, sure, yes, I thought you meant replace the entire pointer with gpointer.

Reply:
My misunderstanding, sorry, agreed.